README.rst

```diff
@@ -63,13 +65,27 @@ Settings list
 -------------
 All settings are contained in a ``PUSH_NOTIFICATIONS_SETTINGS`` dict.
 
-In order to use GCM, you are required to include ``GCM_API_KEY``.
-For APNS, you are required to include ``APNS_CERTIFICATE``.
+In order to use GCM, you are required to include one of ``GCM_API_KEY``, ``GCM_API_KEYS``, or ``GCM_API_KEYS_MODEL``.
+For APNS, you are required to include ``APNS_CERTIFICATE``, ``APNS_CERTIFICATES``, or ``APNS_CERTIFICATES_MODEL``.
 
 - ``APNS_CERTIFICATE``: Absolute path to your APNS certificate file. Certificates with passphrases are not supported.
+- ``APNS_CERTIFICATES``: A dictionary reflecting separate application IDs to separate APNS certificate files.
+- ``APNS_CERTIFICATES_MODEL``: A dictionary containing description of a database model to reflect
```

Comment:
Yes, as I mentioned above, the abstract Application model for the push_notifications application is more appropriate than concrete, just because it is more flexible and applicable for more number of use cases.

The initial reason to have a custom model for the push_notifications application was a backward compatibility with the only settings.py as a source of the certificate(s). If the backward compatibility is not necessary, probably the best solution is an abstract Application model.

I wanted to note here that the custom Application model is not conflicted with abstract base Application class and moreover, complements it as you described here.